### PR TITLE
(wiiu) Use portlibs zlib and libpng

### PR DESCRIPTION
**Pull request by opendata26**
## Description

Includes libpng and switches to the portlibs zlib as libpng depends on it  - this is needed for a few cores. This might need  changes on buildbot to install these libraries, which I can provide instructions for if needed.

## Reviewers

@aliaspider @QuarkTheAwesome @bparker06

### CODE REVIEW

**Comment by QuarkTheAwesome** [comment]
Assuming there's no problems on the buildbot, I've got no problem with this. We can basically guarantee every devkitPPC installation comes with zlib and libpng; some have even more - it shouldn't affect most users.
Alternatively, we could look at adding libpng in source form, similar to how zlib and libiosuhax are handled now - but in the meantime, this is a perfectly good solution. :+1: from me!

**Comment by twinaphex** [comment]
Is the builtin zlib not sufficient or something?

**Comment by opendata26** [comment]
I switched to Portlibs zlib as its a dependency  on basically any distribution that provides a package for Portlibs libpng and if you compile libpng manually you need it for configure to succeed, and normally it's better to use system libraries.

**Comment by twinaphex** [comment]
Which cores is libpng needed for?

**Comment by opendata26** [comment]
Mupen, fbalpha sdl, sdlvice - not official yet

File: Makefile.wiiu [hunk 1]
```diff
@@ -118,7 +118,6 @@ endif
    	HAVE_RGUI = 1
    	HAVE_ZLIB = 1
    	HAVE_7ZIP = 1
-   	HAVE_BUILTINZLIB = 1
```
Comment (aliaspider) [inline Makefile.wiiu]: `HAVE_BUILTINZLIB = 0` instead of removing the line.

**Comment by opendata26** [comment]
@aliaspider done

**Comment by opendata26** [comment]
Any updates on this, is it okay?

**Comment by bparker06** [comment]
>We can basically guarantee every devkitPPC installation comes with zlib and libpng

It's not in our portlibs folder on the buildbot or the toolchain used by travis:

```
devkitpro/portlibs/ppc/lib$ ls -l
total 372
-rw-rw-r-- 1 buildbot buildbot 337722 Jan 26  2017 libfat.a
-rw-rw-r-- 1 buildbot buildbot  39226 Jan 23  2017 libiosuhax.a
```

**Comment by aliaspider** [comment]
no we don't have portlibs installed, just dkp.

**Comment by opendata26** [comment]
Well 99% of all devkitpro installs come with Portlibs, are you open to installing it

**Comment by aliaspider** [comment]
they are ? it wasn't the case with dkp29 which we are using. I think the perl script was used to install it.
they might have added portlibs in later versions.
I have personally no problems with adding it to the bot and linking libpng/libz by default, but that isn't my decision to make.

**Comment by opendata26** [comment]
I mean, 99% of people with devkitpro installed have Portlibs installed as its required for basically any homebrew

**Comment by aliaspider** [comment]
that figure is clearly wrong, I don't have portlibs and it isn't required for retroarch :P.
it would be even generous to assume that 50% of installs have it. this isn't a blocking issue of course since installing portlibs is pretty simple.

**Comment by cucholix** [comment]
@aliaspider , talked with @twinaphex about this PR and he say you can look into bot files:
`cucholix: ali can install the needed packages, but it's up to you greenlight it. Then bparker can check if ali's changes took place.
Autechre - Today at 4:42 PM
you mean on the buildbot?
sure
let ali know he can look into that`

Vìa Discord DM

**Comment by aliaspider** [comment]
ok I take that back, installing portlibs isn't simple anymore, there used to be a github repo for that but that was removed it seems.
is pacman the only way to get portlibs now ? that wouldn't work here since we aren't using latest dkppc.

**Comment by aliaspider** [comment]
ok both zlib and libpng-1.6.21 are installed now on the buildbot.

**Comment by twinaphex** [comment]
OK I will merge the PR now. Hopefully we will not run into too many issues.

**Comment by twinaphex** [comment]
I have to revert this - it is already causing issues for us.

List the cores that require this, I will then shallow fork them and bake the dependencies in on the core side instead, the way it is supposed to. We do not want to require additional dependencies for building REtroArch for WiiU, it should be kept simple.

**Comment by opendata26** [comment]
@twinaphex , mupen64plus mainly

**Comment by twinaphex** [comment]
there is a wiiu core for mupen64plus right now ?

**Comment by aliaspider** [comment]
ok it turns out I made a mistake during configure and ended up with  incorrect libs for the buildbot. it should be fine now.

**Pull request merged**